fpga_interchange/arch.h

```diff
@@ -0,0 +1,1121 @@
+        getTileXY(tile_index, &loc->x, &loc->y);
+    }
+
+    int getGridDimX() const { return chip_info->width; }
+    int getGridDimY() const { return chip_info->height; }
+    int getTileBelDimZ(int x, int y) const {
+        return chip_info->tile_types[chip_info->tiles[getTileIndex(x, y)].type].num_bels;
+    }
+    int getTilePipDimZ(int x, int y) const {
+        return chip_info->tile_types[chip_info->tiles[getTileIndex(x, y)].type].number_sites;
+    }
+
+    // -------------------------------------------------
+
+    void setup_byname() const;
+
+    BelId getBelByName(IdString name) const;
+
+    IdString getBelName(BelId bel) const
+    {
+        NPNR_ASSERT(bel != BelId());
+        int site_index = locInfo(bel).bel_data[bel.index].site;
+        NPNR_ASSERT(site_index >= 0);
+        const SiteInstInfoPOD &site = chip_info->sites[chip_info->tiles[bel.tile].sites[site_index]];
+        return id(std::string(site.name.get()) +
+                "/" + IdString(locInfo(bel).bel_data[bel.index].name).str(this));
+    }
+
+    uint32_t getBelChecksum(BelId bel) const { return bel.index; }
+
+    void bindBel(BelId bel, CellInfo *cell, PlaceStrength strength)
+    {
+        NPNR_ASSERT(bel != BelId());
+        NPNR_ASSERT(tileStatus[bel.tile].boundcells[bel.index] == nullptr);
+
+        tileStatus[bel.tile].boundcells[bel.index] = cell;
+        cell->bel = bel;
+        cell->belStrength = strength;
+        refreshUiBel(bel);
+    }
+
+    void unbindBel(BelId bel)
+    {
+        NPNR_ASSERT(bel != BelId());
+        NPNR_ASSERT(tileStatus[bel.tile].boundcells[bel.index] != nullptr);
+        tileStatus[bel.tile].boundcells[bel.index]->bel = BelId();
+        tileStatus[bel.tile].boundcells[bel.index]->belStrength = STRENGTH_NONE;
+        tileStatus[bel.tile].boundcells[bel.index] = nullptr;
+        refreshUiBel(bel);
+    }
+
+    bool checkBelAvail(BelId bel) const
+    {
+        return tileStatus[bel.tile].boundcells[bel.index] == nullptr;
+    }
+
+    CellInfo *getBoundBelCell(BelId bel) const
+    {
+        NPNR_ASSERT(bel != BelId());
+        return tileStatus[bel.tile].boundcells[bel.index];
+    }
+
+    CellInfo *getConflictingBelCell(BelId bel) const
+    {
+        NPNR_ASSERT(bel != BelId());
+        return tileStatus[bel.tile].boundcells[bel.index];
+    }
+
+    BelRange getBels() const
+    {
+        BelRange range;
+        range.b.cursor_tile = 0;
+        range.b.cursor_index = -1;
+        range.b.chip = chip_info;
+        ++range.b; //-1 and then ++ deals with the case of no Bels in the first tile
+        range.e.cursor_tile = chip_info->width * chip_info->height;
+        range.e.cursor_index = 0;
+        range.e.chip = chip_info;
+        return range;
+    }
+
+    Loc getBelLocation(BelId bel) const
+    {
+        NPNR_ASSERT(bel != BelId());
+        Loc loc;
+        getTileXY(bel.tile, &loc.x, &loc.y);
+        loc.z = bel.index;
+        return loc;
```

Comment:
Without it, the placer will treat arcs from global buffer cells as relevant, even though distance from global buffers (in general) should not count towards the wirelength calculation in the placer in the same way as other arcs.

Reply:
Sorry, I feel like this comment is missing the first sentence?

Reply:
I think I submitted it on the wrong line, sorry - I was referring to the TODO on getBelGlobalBuf